src/overlays/actors/ovl_Obj_Lupygamelift/z_obj_lupygamelift.c

```diff
@@ -28,35 +33,142 @@ const ActorInit Obj_Lupygamelift_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lupygamelift/func_80AF04BC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lupygamelift/func_80AF04D8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lupygamelift/func_80AF0514.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lupygamelift/func_80AF0530.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lupygamelift/ObjLupygamelift_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lupygamelift/ObjLupygamelift_Draw.s")
+void ObjLupygamelift_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjLupygamelift* this = THIS;
+    Path* path;
+    s32 phi_v0;
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    this->dyna.actor.scale.y = 0.15f;
+    this->dyna.actor.shape.rot.x = 0;
+    this->dyna.actor.world.rot.x = 0;
+    this->dyna.actor.shape.rot.z = 0;
+    this->dyna.actor.world.rot.z = 0;
+    this->unk_170 = 0;
+    Actor_UpdateBgCheckInfo(globalCtx, thisx, 0.0f, 0.0f, 0.0f, 4);
+    ActorShape_Init(&thisx->shape, 0.0f, ActorShadow_DrawSquare, 0.0f);
+    DynaPolyActor_Init(&this->dyna, 1);
+    DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &object_raillift_Colheader_0048D0);
+    this->unk_160 = thisx->home.rot.z * 0.1f;
+    if (this->unk_160 < 0.0f) {
+        this->unk_160 = -this->unk_160;
+    }
+    this->dyna.actor.home.rot.x = 0;
+    this->dyna.actor.home.rot.y = 0;
+    this->dyna.actor.home.rot.z = 0;
+    path = &globalCtx->setupPathList[OBJLUPYGAMELIFT_GET_PATH(thisx)];
+    this->unk_168 = OBJLUPYGAMELIFT_GET_7(thisx);
+    this->unk_164 = path->count;
+    if (this->unk_168 >= this->unk_164) {
+        this->unk_168 = 0;
+    }
+    this->unk_16C = Lib_SegmentedToVirtual(path->points);
+    Actor_SpawnAsChild(&globalCtx->actorCtx, &this->dyna.actor, globalCtx, 0x183, this->dyna.actor.world.pos.x,
+                       this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z, this->dyna.actor.shape.rot.x,
+                       this->dyna.actor.shape.rot.y, this->dyna.actor.shape.rot.z, 0);
+    if (OBJLUPYGAMELIFT_GET_C(thisx) != 0) {
+        phi_v0 = 1;
+    } else {
+        phi_v0 = 0;
+    }
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, 0x1D2, this->dyna.actor.home.pos.x, this->dyna.actor.home.pos.y,
+                this->dyna.actor.home.pos.z, 0, 0, 0, phi_v0);
+    func_80AF04BC(this);
+}
+
+void ObjLupygamelift_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjLupygamelift* this = THIS;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80AF0394(ObjLupygamelift* this) {
+    static f32 D_80AF0750 = 100.0f;
+    static f32 D_80AF0754 = 255.0f;
+    static f32 D_80AF0758 = 5.0f;
+    static f32 D_80AF075C = 10.0f;
+    static f32 D_80AF0760 = -240.0f;
+    static f32 D_80AF0764 = 30.0f;
+
+    f32 new_var = D_80AF0754 - D_80AF0750;
```

Comment:
```suggestion
    static f32 D_80AF0764 = 30.0f;
    f32 new_var = D_80AF0754 - D_80AF0750;
```
No space needed here.